Assets/Scripts/PoolSystem/PoolSystem.cs

```diff
@@ -0,0 +1,122 @@
+        public void SpawnObject(string nameTag, GameObject spawnPrefab, Transform parentTransform = null) 
+            => SpawnObject(nameTag, parentTransform, spawnPrefab);
+
+        /// <summary>
+        /// Creates new object if there aren't objects in the pool or put objects from the pool into the scene.
+        /// </summary>
+        /// <param name="nameTag">The name of the category the object is in.</param>
+        /// <param name="parentTransform">The object you want the spawnobject to be the childobject of.</param>
+        /// <param name="spawnPrefab">The object you want to spawn in.</param>
+        public void SpawnObject(string nameTag, Transform parentTransform = null, GameObject spawnPrefab = null)
+        {
+            if (_objectQueu?.ContainsKey(nameTag) == true && _objectQueu[nameTag].Any())
+            {
+                GameObject lastObject = _objectQueu[nameTag].LastOrDefault();
+                _objectQueu[nameTag].Remove(lastObject);
+
+                MoveObjectToActiveScene(lastObject, parentTransform);
+                AddActiveObject(nameTag, lastObject);
+
+                lastObject.SetActive(true);
+                return;
+            }
+
+            if (spawnPrefab == null) 
+                return;
+
+            GameObject newGameObject = Instantiate(spawnPrefab);
+            MoveObjectToActiveScene(newGameObject, parentTransform);
+            AddActiveObject(nameTag, newGameObject);
+        }
+
+        /// <summary>
+        /// Puts the given object in the poolsystem.
+        /// </summary>
+        /// <param name="nameTag">The name of the category the object should be in.</param>
+        /// <param name="despawnObject">The object you want to despawn.</param>
+        public void DespawnObject(string nameTag, GameObject despawnObject)
+        {
+            if (!(_objectsActive.ContainsKey(nameTag) && _objectsActive[nameTag].Contains(despawnObject)))
+            {
+                Debug.LogError($"Can't find object {nameTag} to despawn");
+                return;
+            }
+
+            List<GameObject> objects = _objectsActive[nameTag];
+
+            MoveObjectToDontDestroyOnLoadScene(despawnObject);
+            AddQueuObject(nameTag, despawnObject);
+
+            despawnObject.SetActive(false);
+            objects.Remove(despawnObject);
+        }
+
+        /// <summary>
+        /// Starts coroutine DeathTimer
+        /// </summary>
+        /// <param name="respawnTime">How long do you want the death timer to last</param>
+        /// <param name="objectName">Name of the object you want to bring back after the timer</param>
+        /// <param name="previousParentTransform">The parent object you want the object to appear under</param>
+        public void StartRespawnTimer(int respawnTime, string objectName, Transform previousParentTransform) 
+            => StartCoroutine(RespawnTimer(respawnTime, objectName ,previousParentTransform));
```

Comment:
```suggestion
            => StartCoroutine(RespawnTimer(respawnTime, objectName, previousParentTransform));
```
Sorry my ocd cant handle